packages/mapviewer/src/utils/offline/OfflineReadinessStatus.vue

```diff
@@ -0,0 +1,167 @@
+            cache: 'no-store',
+            headers: {
+                cache: 'no-store',
+                'cache-control': 'no-cache',
+            },
+        })
+
+        if (resp?.status === 200) {
+            await registration.update()
+        }
+    }, period)
+}
+
+const { offlineReady, needRefresh, updateServiceWorker } = useRegisterSW({
+    immediate: true,
+    onRegisteredSW(serviceWorkerUrl, registration) {
+        if (registration?.active?.state === 'activated') {
+            isServiceWorkerActive.value = true
+            registerPeriodicSync(serviceWorkerUrl, registration)
+        } else if (registration?.installing) {
+            registration.installing.addEventListener('statechange', (e) => {
+                const sw = e.target as ServiceWorker
+                isServiceWorkerActive.value = sw.state === 'activated'
+                if (isServiceWorkerActive.value) {
+                    registerPeriodicSync(serviceWorkerUrl, registration)
+                }
+            })
+        }
+    },
+})
+
+const statusIcon = computed<string>(() => {
+    if (offlineReady.value) {
+        return 'check'
+    }
+    if (needRefresh.value) {
+        return 'triangle-exclamation'
+    }
+    return 'circle-notch'
+})
+const shouldStatusIconSpin = computed<boolean>(() => !offlineReady.value && !needRefresh.value)
+const tooltipContent = computed<string>(() => {
+    const title = t('offline_modal_title')
+    let extraInfoKey = 'wait_data_loading'
+    if (needRefresh.value) {
+        extraInfoKey = 'offline_cache_obsolete'
+    }
+    if (offlineReady.value) {
+        extraInfoKey = 'offline_dl_succeed'
+    }
+    return `${title}: ${t(extraInfoKey)}`
+})
+
+function refreshCache() {
+    updateServiceWorker(true)
+        .then(() => {
+            log.info({
+                title: 'OfflineReadinessStatus component',
+                titleColor: LogPreDefinedColor.Sky,
+                messages: ['ServiceWorker cache refreshed'],
+            })
+            window.location.reload()
+        })
+        .catch((error) => {
+            log.error({
+                title: 'OfflineReadinessStatus component',
+                titleColor: LogPreDefinedColor.Sky,
+                messages: ['Failed to refresh the ServiceWorker cache', error],
+            })
+        })
+}
+
+// #v-endif
+</script>
+
+<template>
+    <GeoadminTooltip :tooltip-content="tooltipContent">
+        <div class="tw:flex tw:justify-around tw:items-center tw:p-1">
+            <FontAwesomeLayers>
+                <FontAwesomeIcon
+                    icon="slash"
+                    mask="plug"
+                />
+                <FontAwesomeIcon
+                    icon="slash"
+                    class="tw:text-red-600"
+                    transform="up-1"
+                />
+            </FontAwesomeLayers>
+            <span
+                v-if="withText"
+                class="fw-bold mx-1"
```

Comment:
You could maybe also use [gap](https://tailwindcss.com/docs/gap) in the enclosing flexbox